src/pygeoprocessing/routing/routing.pyx

```diff
@@ -4159,6 +4216,131 @@ def calculate_subwatershed_boundary(
+
+    Example:
+        raster A contains the following
+            * pixel at (3, 4) at 10m draining to a nodata  pixel
+            * pixel at (15, 19) at 11m draining to a nodata pixel
+            * pixel at (19, 21) at 10m draining to a nodata pixel
+            * pit pixel at (10, 15) at 5m surrounded by non-draining pixels
+            * pit pixel at (25, 15) at 15m surrounded by non-draining pixels
+            * pit pixel at (2, 125) at 5m surrounded by non-draining pixels
+
+        The result is two pixels indicating the first lowest edge and first
+        lowest sink seen:
+            drain_pixel = (3, 4), 10
+            sink_pixel = (10, 15), 5
+
+    Args:
+        dem_raster_path_band (tuple): a raster/path band tuple to detect
+            sinks in.
+
+    Return:
+        (drain_pixel, drain_height, sink_pixel, sink_height) -
+            two (x, y) tuples with corresponding heights, first
+            list is for edge drains, the second is for pit sinks. The x/y
+            coordinate is in raster coordinate space and `height` is the
+            height of the given pixels in edge and pit respectively.
+    """
+    # this outer loop drives the raster block search
+    cdef double lowest_drain_height = numpy.inf
+    cdef double lowest_sink_height = numpy.inf
+
+    drain_pixel = None
+    sink_pixel = None
+
+    dem_raster_info = pygeoprocessing.get_raster_info(
+        dem_raster_path_band[0])
+    cdef double dem_nodata
+    # guard against undefined nodata by picking a value that's unlikely to
+    # be a dem value
+    if dem_raster_info['nodata'][0] is not None:
+        dem_nodata = dem_raster_info['nodata'][0]
+    else:
+        dem_nodata = IMPROBABLE_FLOAT_NODATA
+
+    raster_x_size, raster_y_size = dem_raster_info['raster_size']
+
+    cdef _ManagedRaster dem_managed_raster = _ManagedRaster(
+        dem_raster_path_band[0], dem_raster_path_band[1], 0)
+
+    cdef time_t last_log_time = ctime(NULL)
+
+    for offset_dict in pygeoprocessing.iterblocks(
+            dem_raster_path_band, offset_only=True, largest_block=0):
+        win_xsize = offset_dict['win_xsize']
+        win_ysize = offset_dict['win_ysize']
+        xoff = offset_dict['xoff']
+        yoff = offset_dict['yoff']
+
+        if ctime(NULL) - last_log_time > _LOGGING_PERIOD:
+            last_log_time = ctime(NULL)
+            current_pixel = xoff + yoff * raster_x_size
+            LOGGER.info(
+                '(infer_sinks): '
+                f'{current_pixel} of {raster_x_size * raster_y_size} '
+                'pixels complete')
+
+        # search block for locally undrained pixels
+        for yi in range(1, win_ysize+1):
+            yi_root = yi-1+yoff
+            for xi in range(1, win_xsize+1):
+                xi_root = xi-1+xoff
+                # this value is set in case it turns out to be the root of a
+                # pit, we'll start the fill from this pixel in the last phase
+                # of the algorithm

```

Comment:
Update this comment? I don't think we're doing any filling here.

Reply:
Removed in 1e3f6bf